framework/include/interfaces/PerfGraphInterface.h

```diff
@@ -0,0 +1,61 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef PERFGRAPHINTERFACE_H
+#define PERFGRAPHINTERFACE_H
+
+#include "Moose.h"
+#include "PerfGuard.h"
+#include "InputParameters.h"
+
+#define TIME_SECTION(id) PerfGuard time_guard(_perf_graph, id);
```

Comment:
The macro approach is good because one of the other features I'd like to see (especially during development), is the ability to remove all calls to timing with #define statements like we do for mooseAssert().

Reply:
I'll add the ability to turn this off with a `-D`.  How about `-DMOOSE_NO_PERF_GRAPH`?  And since we don't have configure (😉) I guess I'll make a new environment variable to turn it on?